In [1]:
###Hypothesis Testing
import numpy as np
from scipy import stats

In [9]:
# sample data
sample_mean = 3050
n = 25

# theoretical model
X = 600
mu = 1000 + 5 * X

# standard deviation
sigma = 5 * 25

print("Theoretical Mean:", mu)
print("Sigma:", sigma)

Theoretical Mean: 4000
Sigma: 125


Null Hypothesis (H0):
The weekly operating cost follows the theoretical model.

H0 : μ = 4000

Alternative Hypothesis (H1):
The weekly operating cost is higher than the theoretical model.

H1 : μ > 4000

Calculate Test Statistic (Z)

    Formula

𝑍=(𝑥ˉ−𝜇)/(𝜎/𝑛)
Z=(xˉ−μ)/(σ/n)

In [10]:
z = (sample_mean - mu) / (sigma / np.sqrt(n))
print("Z Statistic:", z)

Z Statistic: -38.0


In [12]:
alpha = 0.05

critical_value = stats.norm.ppf(1 - alpha)

print("Critical Value:", critical_value)

Critical Value: 1.6448536269514722


In [13]:
###Decision rule
if z > critical_value:
    print("Reject Null Hypothesis")
else:
    print("Fail to Reject Null Hypothesis")

Fail to Reject Null Hypothesis


Conclusion

The calculated Z statistic is -38.0, which is less than the critical value 1.645.

Therefore, we fail to reject the null hypothesis.

There is no strong statistical evidence to support the restaurant owners' claim that the weekly operating costs are higher than predicted by the theoretical model.

In [14]:
###Confidence Interval

In [15]:
import numpy as np
from scipy.stats import t, norm

In [16]:
data = [1.13,1.55,1.43,0.92,1.25,1.36,1.32,0.85,1.07,1.48,1.20,1.33,1.18,1.22,1.29]

n = len(data)

mean = np.mean(data)

std = np.std(data, ddof=1)

print("Sample Mean:", mean)
print("Sample Standard Deviation:", std)

Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936


In [17]:
###Sample SD
alpha = 0.01

t_value = t.ppf(1 - alpha/2, df=n-1)

margin_error = t_value * (std / np.sqrt(n))

lower = mean - margin_error
upper = mean + margin_error

print("99% CI using Sample SD:")
print(lower, upper)

99% CI using Sample SD:
1.0901973384384906 1.3871359948948425


In [20]:
###Population SD
sigma = 0.2

z_value = norm.ppf(1 - alpha/2)

margin_error = z_value * (sigma / np.sqrt(n))

lower = mean - margin_error
upper = mean + margin_error

print("99% CI using Population SD:")
print(lower, upper)

99% CI using Population SD:
1.1056514133957607 1.3716819199375725
